File: Solutions/2215. Find the Difference of Two Arrays/2215. Find the Difference of Two Arrays.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> s1(nums1.begin(),nums1.end()),s2(nums2.begin(),nums2.end());
        vector<vector<int>> res(2);
        for(auto& n:s1){
            if(s2.count(n)==0){
                res[0].push_back(n);
            }
        }
        for(auto& n:s2){
            if(s1.count(n)==0){
                res[1].push_back(n);
            }
        }
        return res;
    }
};

File: Solutions/2215. Find the Difference of Two Arrays/2215. Find the Difference of Two Arrays.go
func findDifference(nums1 []int, nums2 []int) [][]int {
    s1,s2:=make(map[int]bool),make(map[int]bool)
    for _,n:=range(nums1){
        s1[n]=true
    }
    for _,n:=range(nums2){
        s2[n]=true
    }
    res:=make([][]int,2)
    for k,_:=range s1{
        if _,ok:=s2[k];!ok{
            res[0]=append(res[0],k)
        }
    }
    for k,_:=range s2{
        if _,ok:=s1[k];!ok{
            res[1]=append(res[1],k)
        }
    }
    return res
}